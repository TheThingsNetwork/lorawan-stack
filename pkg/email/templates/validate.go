// Copyright Â© 2022 The Things Network Foundation, The Things Industries B.V.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package templates

import (
	"time"

	"go.thethings.network/lorawan-stack/v3/pkg/email"
	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
)

func init() {
	tmpl, err := email.NewTemplateFS(
		fsys, ttnpb.NotificationType_VALIDATE,
		email.FSTemplate{
			SubjectTemplate:      "Please confirm your email address for {{ .Network.Name }}",
			HTMLTemplateBaseFile: "base.html.tmpl",
			HTMLTemplateFile:     "validate.html.tmpl",
			TextTemplateFile:     "validate.txt.tmpl",
		},
	)
	if err != nil {
		panic(err)
	}
	email.RegisterTemplate(tmpl)
}

// ValidateData is the data for the validate email.
type ValidateData struct {
	email.TemplateData
	*ttnpb.EntityIdentifiers
	ID    string
	Token string
	TTL   time.Duration
}
