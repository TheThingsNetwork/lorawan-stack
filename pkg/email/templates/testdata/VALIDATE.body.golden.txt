Hello,

Please confirm your email address for The Things Network.

Your email address will be linked to user "foo-usr".

You can go to https://eu1.cloud.thethings.network/oauth/validate?reference=ID&token=TOKEN to confirm your email address.
Alternatively, you can use the reference "ID" and confirmation token "TOKEN" directly.

The confirmation token expires an hour from now, so if you don't confirm your email address before then, you'll have to request a new confirmation if you still want to confirm your email address.
